app/policies/book_policy.rb

```diff
@@ -22,6 +22,14 @@ def update?
     record.user == user
   end
 
+  def reserve?
+    record.user != user
+  end
+
+  def unreserve?
+    record.borrower_id == user.id && record.user_id != user.id
```

Comment:
I don't think we need `record.user_id != user.id` (as this scenario should never happen, since a user cannot reserve their own books).

Reply:
we sorted it out on Friday:)